Classes/PKStateMachine.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  PKStateMachine.swift
+//  Pods
+//
+//  Created by Gal Orlanczyk on 02/04/2017.
+//
+//
+
+import Foundation
+
+protocol IntRawRepresentable: RawRepresentable {
+    var rawValue: Int { get }
+}
+
+protocol StateProtocol: IntRawRepresentable, Hashable {}
```

Comment:
Why call it StateProtocol rather than State (or PKState)? AFAIK the "Protocol" suffix is only needed for legacy ObjC protocols, to avoid collision with classes.